### PR TITLE
Support docker build --build-context parameter

### LINKED ISSUE #3271: Allow the passing of --build-context args to docker build
I would like to be able to reference files outside of the local Docker build context from within my Dockerfile.tmpl file, using the `docker build --build-context` flag.

https://github.com/docker/buildx/blob/master/docs/reference/buildx_build.md#build-context

We are currently working around this by pulling files from git, however this is unsatisfactory since it requires the files to be checked in first.

### PR DESCRIPTION
**Pull request by jonnyry**
# What does this change

Add new `--build-context` parameter to porter, and pass through the value to `docker build`.

https://github.com/docker/buildx/blob/master/docs/reference/buildx_build.md#build-context

# What issue does it fix
Closes #3271

# Checklist
- [x] Did you write tests?
- [x] Did you write documentation?

### CODE REVIEW

File: pkg/build/buildkit/buildx_test.go [hunk 1]
```diff
@@ -33,6 +34,26 @@ func Test_parseBuildArgs(t *testing.T) {
 	}
 }
 
+func Test_parseBuildContexts(t *testing.T) {
+	testcases := []struct {
+		name      string
+		inputArgs []string
+		wantArgs  map[string]buildx.NamedContext
+	}{
+		{name: "valid args", inputArgs: []string{"A=1", "B=2=2", "C="},
```
Comment (schristoff) [inline pkg/build/buildkit/buildx_test.go:43]: Just checking, are these all valid from Dockers side? I figure if you pass in "C=" and Docker tries to run it it's gonna error, no?

**Review by schristoff — APPROVED** [review]
Some small comment but otherwise great work, thank you

File: pkg/build/buildkit/buildx_test.go (hunk 1 above)
Comment (jonnyry) [reply pkg/build/buildkit/buildx_test.go]: Ah yes good point, "C=" is not.

I've taken another look and found docker buildkit has its own [`ParseContextNames`](https://github.com/docker/buildx/blob/master/util/buildflags/context.go) function with error handling (similar to the `--ssh` param parsing), so I've replaced my hand rolled `parseBuildContexts` with it, so we get dockers own error handling on the parameter.



This yields the following outputs from porter:

```
$ ./porter build --build-context
flag needs an argument: --build-context

$ ./porter build --build-context context1
Copying porter runtime ===> 
Copying mixins ===> 
Copying mixin exec ===> 
Copying mixin terraform ===> 
Copying mixin az ===> 
Building bundle image
error parsing the --build-context flags: invalid context value: context1, expected key=value
unable to build bundle image: error parsing the --build-context flags: invalid context value: context1, expected key=value
unable to build bundle image: error parsing the --build-context flags: invalid context value: context1, expected key=value


$ ./porter build --build-context context1=
Copying porter runtime ===> 
Copying mixins ===> 
Copying mixin exec ===> 
Copying mixin terraform ===> 
Copying mixin az ===> 
Building bundle image
[+] Building 0.0s (0/0)                                                                                                                                                                                                                     
error building docker image: failed to get build context context1: stat : no such file or directory
unable to build bundle image: error building docker image: failed to get build context context1: stat : no such file or directory
unable to build bundle image: error building docker image: failed to get build context context1: stat : no such file or directory


./porter build --build-context context1=..
(command successful)
```

**Review by schristoff — APPROVED** [review]

**Comment by schristoff** [comment]
Good catch, thanks for that update :)

**Pull request merged**